### ISSUE TITLE
issue with URL parameters, cookie mode

### ISSUE THREAD

**Issue by atomGit** [jankkm/invidioucious#9]
ex: https://www.youtube.com/channel/UCcem9I78ybZLHLRUlkUO3sw

v0.9.8.3

if you disable cookie mode, enable dark mode, refresh, then disable dark mode, the `dark_mode=true` remains after refreshing the page

prior ro 0.9.8.3, the URL params worked properly (except for the `?` bug), but they would remain after disabling all options

they remain now as well, but also, once dark mode is enabled, it cannot be disabled (the param is always `dark_mode=true`) regardless of cookie mode state or dark mode state

note that i didn't test other options

**Comment by jankkm** [comment]
Okay, I see. The way the extension works, arguments are only appended if there are no arguments in the url. Do you think the scenario you described actually happens? Maybe if you save the url with arguments as a bookmark.
if, in your described situation, you click on any video in the channel or on the Invidious logo, the new page is loaded with your new settings (dark_mode=false).

**Comment by atomGit** [comment]
i don't think you understand - once dark mode is enabled, it can never be disabled

there are problems with the code

i was going to run test just now, so i opened a tab to go to youtube.com which i knew would be forwarded to whatever invidious instance i had set

as it turns out, i didn't go anywhere (problem loading page) and the URL in my address bar is...

`https://invidious.13ad.de/?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US`

it seems like you're coding in options and then never testing them, or passing off bugs as scenarios that are unlikely to happen

**Comment by jankkm** [comment]
> i don't think you understand - once dark mode is enabled, it can never be disabled

Then please explain it to me. I am not able to produce a situation where darkmode can not be disabled. For me it always works as expected.

> i was going to run test just now, so i opened a tab to go to youtube.com which i knew would be forwarded to whatever invidious instance i had set

> as it turns out, i didn't go anywhere (problem loading page) and the URL in my address bar is...

> https://invidious.13ad.de/?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US?quality=hd720&hl=en-US

That happens if quality=* is the first argument, it does not happen if darkmode or thinmode or someting else is enabled. It will be fixed in the next release.

> it seems like you're coding in options and then never testing them,
how so?

> or passing off bugs as scenarios that are unlikely to happen
Are you criticizing me for prioritizing bugs and working on the more important ones first? I don't even have enough time for those.
